### PR TITLE
Recommended changes done and added 'remove' command

### PR DESCRIPTION
**Pull request by nsn39**
Type in --select flag before the LOCATION argument when using 'air search' command so that user doesn't need to go through the whole table.

### CODE REVIEW

File: air_quality_cli/commands/search.py [hunk 1]
```diff
@@ -2,28 +2,61 @@
...
     show_table,
+    show_menu,
+    get_stations,
 )
 
 
 @click.command()
 @click.argument("query")
-def search(query: str):
+@click.option("--select", is_flag=True)
```
Comment (yankeexe) [inline air_quality_cli/commands/search.py]: Add shorthand for the `--select` option. 

```python 
@click.option("-s", "--select", is_flag=True)
```

File: air_quality_cli/commands/search.py [hunk 2]
```diff
@@ -2,28 +2,61 @@
...
     data = make_request(query)
 
     # Get aqi value and station name from the response payload.
     api_data = get_aqi_data(data, query)
 
-    # Create an info payload that resonates to the aqi values extracted.
-    air_info = create_table_payload(api_data)
+    # Show only one station's data if called with --select flag
+    if select:
+        click.echo("Asked with search")
```
Comment (yankeexe) [inline air_quality_cli/commands/search.py]:
```python
click.echo("Select Station")
```

File: air_quality_cli/commands/search.py [hunk 3]
```diff
@@ -2,28 +2,61 @@
...
+        # Allow user to select station to add to the config file.
+        selected_station = (
+            show_menu(stations) if len(stations) > 1 else stations[0]
+        )
+
+        # Declare a list of tuple to hold the tuple of single station.
+        single_station_list: List[Tuple[str, str]] = []
+
+        for item in api_data:
+            if item[0] == selected_station:
```
Comment (yankeexe) [inline air_quality_cli/commands/search.py]: it's a NamedTuple, you can take the value of station using `item.station`


```python 
if item.station == selected_station:
    ...
```

File: air_quality_cli/commands/search.py [hunk 4]
```diff
@@ -2,28 +2,61 @@
...
+        click.echo("Asked with search")
+
+        # Using simple-term-menu to show the list of stations
+        # Get the name of stations and their uid from the response data.
+        stations_data = get_stations(data)
+
+        # Filter out just the stations.
+        stations = [data.station for data in stations_data]
+
+        # Allow user to select station to add to the config file.
```
Comment (yankeexe) [inline air_quality_cli/commands/search.py]: Allow user to select station to display data received from it.

File: README.md [hunk 5]
```diff
@@ -44,6 +44,12 @@ $ air search kathmandu
 $ air search Nepal
 ```
 
+You can also use --select flag to select a particular station to display details.
+
+```bash
+$ air search --select Nepal
```
Comment (yankeexe) [inline README.md:50]: Add another example with the shorthand example. 

```bash 
$ air search -s Nepal
```

File: air_quality_cli/commands/search.py [hunk 6]
```diff
@@ -2,28 +2,61 @@
...
+    # Show only one station's data if called with --select flag
+    if select:
+        click.echo("Asked with search")
+
+        # Using simple-term-menu to show the list of stations
+        # Get the name of stations and their uid from the response data.
+        stations_data = get_stations(data)
+
+        # Filter out just the stations.
+        stations = [data.station for data in stations_data]
```
Comment (yankeexe) [inline air_quality_cli/commands/search.py]: type hint here: 

`stations: List[str]`

**Comment by yankeexe** [comment]
@nsn39 Thank you for the contribution. Can you please go through the reviews, make the changes 🙂

File: README.md (hunk 5 above)
Comment (yankeexe) [reply README.md:50]: @nsn39 Please address this comment as well.

**Comment by yankeexe** [comment]
@nsn39  I see you have added the `remove` command as well, if you read the issue #3 to properly address the removing of stations. 

We need to store `uid` along with `station_name` as a Tuple for the search query key. So that we can show a list of stations to allow users to choose which station to delete. 

> It will help in the regard that, we don't have to make an API request to get the name of the station that has to be deleted.

File: air_quality_cli/constants.py [hunk 7]
```diff
@@ -63,7 +65,7 @@
         ),
     ),
     (
-        range(301, 501),
+        range(301, 1001),
```
Comment (yankeexe) [inline air_quality_cli/constants.py:68]: @nsn39  Can you share the resource, where the range from 300-1000 is considered hazardous? For most, the cap was 500.

**Reference:**  [Indices by location](https://en.wikipedia.org/wiki/Air_quality_index#Indices_by_location)

Comment (nsn39) [reply air_quality_cli/constants.py:68]: aqi_value1
 This is the place where the aqi exceeds 500.

**Comment by nsn39** [comment]
Now, the stations are saved in a list of dicts with attributes 'name' and 'uid' to for name and uid. About the place where the AQI was exceeding expected cap is shown in the screenshot above. It was the only location I found where AQI was higher than 500.

**Comment by yankeexe** [comment]
@nsn39 application has been rewritten in Go. Closing this, thank you for the contribution. :slightly_smiling_face:

**Pull request closed without merging**